packages/dm-core-plugins/src/form/templates/shared/PrimitiveArray.tsx

```diff
@@ -0,0 +1,122 @@
+import { Button, Icon, Tooltip } from '@equinor/eds-core-react'
+import { add } from '@equinor/eds-icons'
+import React, { useState } from 'react'
+import { AttributeField } from '../../fields/AttributeField'
+import { TUiAttributeObject } from '../../types'
+import { DeleteSoftButton, TAttribute } from '@development-framework/dm-core'
+
+interface PrimitiveArrayProps {
+  uiAttribute: TUiAttributeObject | undefined
+  attribute: TAttribute
+  data: any
+  namePath: string
+  onChange: (data: any[]) => void
+}
+
+const getDefaultValue = (type: string): string | boolean | number => {
+  switch (type) {
+    case 'boolean':
+      return true
+    case 'number':
+      return 0
+    default:
+      return ''
+  }
+}
+
+const PrimitiveArray = ({
+  data,
+  attribute,
+  uiAttribute,
+  namePath,
+  onChange,
+}: PrimitiveArrayProps) => {
+  const [hovering, setHovering] = useState<number>(-1)
+
+  const updateValues = (
+    index: number,
+    newValue: boolean | string | number
+  ): void => {
+    const newValues = [...data]
+    switch (attribute.attributeType) {
+      case 'boolean':
+        newValues[index] = newValue
+        onChange(newValues)
+        break
+      case 'number':
+        newValues[index] = Number(newValue) ?? 0
+        onChange(newValues)
+        break
+      default:
+        newValues[index] = newValue
+        onChange(newValues)
+    }
+  }
+  const removeItem = (index: number) => {
+    const newValues = [...data]
+    newValues.splice(index, 1)
+    onChange(newValues)
+  }
+
+  return (
+    <div
+      className='
+    flex
+    flex-col
+    max-h-96
+    w-fit
+    content-start
+    overflow-y-auto
+    overflow-x-hidden
+    pb-2
+  '
+    >
+      {data.map((item: any, index: number) => (
```

Comment:
Shouldn't use `any` unless absolutely necessary. I'm guessing the proper type would be:

```suggestion
      {data.map((item: string | number | boolean, index: number) => (
```

Reply:
Done, replaced with type Primitive